backend_new/src/services/user.service.ts

```diff
@@ -0,0 +1,234 @@
+            },
+          },
+          {
+            listings: {
+              some: {
+                name: {
+                  contains: params.search,
+                  mode: Prisma.QueryMode.insensitive,
+                },
+              },
+            },
+          },
+        ],
+      });
+    }
+
+    if (!params.filter?.length) {
+      return {
+        AND: filters,
+      };
+    }
+
+    params.filter.forEach((filter) => {
+      if (UserFilterKeys.isPartner in filter) {
+        filters.push({
+          userRoles: {
+            isPartner: filter[UserFilterKeys.isPartner],
+          },
+        });
+      } else if (UserFilterKeys.isPortalUser in filter) {
+        if (filter[UserFilterKeys.isPortalUser]) {
+          if (user?.userRoles?.isAdmin) {
+            filters.push({
+              OR: [
+                {
+                  userRoles: {
+                    isPartner: true,
+                  },
+                },
+                {
+                  userRoles: {
+                    isAdmin: true,
+                  },
+                },
+                {
+                  userRoles: {
+                    isJurisdictionalAdmin: true,
+                  },
+                },
+              ],
+            });
+          } else if (user?.userRoles?.isJurisdictionalAdmin) {
+            filters.push({
+              OR: [
+                {
+                  userRoles: {
+                    isPartner: true,
+                  },
+                },
+                {
+                  userRoles: {
+                    isJurisdictionalAdmin: true,
+                  },
+                },
+              ],
+            });
+            filters.push({
+              jurisdictions: {
+                some: {
+                  id: {
+                    in: user?.jurisdictions?.map((juris) => juris.id),
+                  },
+                },
+              },
+            });
+          } else {
+            filters.push({
+              userRoles: {
+                isPartner: true,
+              },
+            });
+          }
+        } else {
+          filters.push({
+            AND: [
+              {
+                OR: [
+                  {
+                    userRoles: {
+                      isPartner: null,
+                    },
+                  },
+                  {
+                    userRoles: {
+                      isPartner: false,
+                    },
+                  },
+                ],
+              },
+              {
+                OR: [
+                  {
+                    userRoles: {
+                      isAdmin: null,
+                    },
+                  },
+                  {
+                    userRoles: {
+                      isAdmin: false,
+                    },
+                  },
+                ],
+              },
+            ],
+          });
+        }
+      }
+    });
```

Comment:
recommend seeing if you can clean this up to not require nested conditionals.

Reply:
I moved some stuff around, lemme know what you think

Reply:
Hey @YazeedLoonat , I see that there are two ways to filter for if a user is a partner, by `UserFilterKeys.isPartner` and `UserFilterKeys.isPortalUser` and then adding to the filter `userRoles.isPartner`. Can we align on one of those approaches instead of using both?

Reply:
after digging through some code I think both of those conditions are unneeded went ahead and cleaned that up!